api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/DetermineEntitlementMessageProcessor.java

```diff
@@ -57,36 +57,52 @@ public MessageType supportsMessageType() {
     @Transactional(Transactional.TxType.REQUIRES_NEW)
     public MessageStatus processMessage(Message message) {
         DetermineEntitlementMessageContext messageContext = messageContextLoader.loadDetermineEntitlementContext(message);
-        Claimant claimant = messageContext.getClaim().getClaimant();
+        Claim claim = messageContext.getClaim();
         PaymentCycle currentPaymentCycle = messageContext.getCurrentPaymentCycle();
         PaymentCycle previousPaymentCycle = messageContext.getPreviousPaymentCycle();
 
         EligibilityAndEntitlementDecision decision = eligibilityAndEntitlementService.evaluateExistingClaimant(
-                claimant,
+                claim.getClaimant(),
                 currentPaymentCycle.getCycleStartDate(),
                 previousPaymentCycle);
 
         paymentCycleService.updatePaymentCycle(currentPaymentCycle, decision);
+        handleDecision(claim, currentPaymentCycle, decision);
+        return COMPLETED;
+    }
 
+    private void handleDecision(Claim claim, PaymentCycle currentPaymentCycle, EligibilityAndEntitlementDecision decision) {
         if (decision.getEligibilityStatus() == ELIGIBLE) {
             createMakePaymentMessage(currentPaymentCycle);
+        } else if (decision.noChildrenPresentInCurrentCycle() && claimantIsNotPregnant(claim.getClaimant())) {
+            handleNoLongerEligibleForScheme(claim);
         } else {
-            handleNonEligibleClaim(currentPaymentCycle);
+            handleNonEligibleQualifyingBenefitStatusClaim(claim);
         }
+    }
 
-        return COMPLETED;
+    private boolean claimantIsNotPregnant(Claimant claimant) {
+        return claimant.getExpectedDeliveryDate() == null;
```

Comment:
Needs to be updated to take into account claims with pregnancy but are too far in the past.

Reply:
Done, we now use PregnancyEntitlementCalculator to do that check.